plugins/modules/identity/keycloak/keycloak_client_scopemapping.py

```diff
@@ -0,0 +1,333 @@
+                    name:
+                        description: Name of the role
+                        type: str
+                    id:
+                        description: ID of the role
+                        type: str
+
+extends_documentation_fragment:
+- community.general.keycloak
+
+
+author:
+    - Fynn Chen (@fynncfchen)
+'''
+
+EXAMPLES = '''
+- name: Map a client role to a client, authentication with credentials
+  community.general.keycloak_client_scope_mapping:
+    realm: MyCustomRealm
+    auth_client_id: admin-cli
+    auth_keycloak_url: https://auth.example.com/auth
+    auth_realm: master
+    auth_username: USERNAME
+    auth_password: PASSWORD
+    state: present
+    client_id: client1
+    client_role:
+        client_id: client2
+        roles:
+          - name: role_name1
+            id: role_id1
+          - name: role_name2
+            id: role_id2
+  delegate_to: localhost
+
+- name: Map a client role to a client, authentication with token
+  community.general.keycloak_client_scope_mapping:
+    realm: MyCustomRealm
+    auth_client_id: admin-cli
+    auth_keycloak_url: https://auth.example.com/auth
+    token: TOKEN
+    state: present
+    id: 6e1d65de-f01a-4d3a-b7c6-3d581165966f
+    client_role:
+        id: 6e1d65de-f01a-4d3a-b7c6-3d581165966f
+        roles:
+          - name: role_name1
+            id: role_id1
+          - name: role_name2
+            id: role_id2
+  delegate_to: localhost
+
+- name: Unmap client role from a client
+  community.general.keycloak_client_scope_mapping:
+    realm: MyCustomRealm
+    auth_client_id: admin-cli
+    auth_keycloak_url: https://auth.example.com/auth
+    auth_realm: master
+    auth_username: USERNAME
+    auth_password: PASSWORD
+    state: absent
+    id: 6e1d65de-f01a-4d3a-b7c6-3d581165966f
+    client_role:
+        id: 52c4d786-b790-4570-9fc2-037aee19a2c2
+        roles:
+          - name: role_name3
+            id: role_id3
+  delegate_to: localhost
+'''
+
+RETURN = '''
+msg:
+    description: Message as to what action was taken.
+    returned: always
+    type: str
+    sample: "Role role1 assigned to client client1."
+
+proposed:
+    description: Representation of proposed client role mapping.
+    returned: always
+    type: dict
+    sample: {
+      clientId: "test"
+    }
+
+existing:
+    description:
+      - Representation of existing client role mapping.
+      - The sample is truncated.
+    returned: always
+    type: dict
+    sample: {
+        "adminUrl": "http://www.example.com/admin_url",
+        "attributes": {
+            "request.object.signature.alg": "RS256",
+        }
+    }
+
+end_state:
+    description:
+      - Representation of client role mapping after module execution.
+      - The sample is truncated.
+    returned: on success
+    type: dict
+    sample: {
+        "adminUrl": "http://www.example.com/admin_url",
+        "attributes": {
+            "request.object.signature.alg": "RS256",
+        }
+    }
```

Comment:
Is these sample actually true? The seem cut-n-pasted from a different function.  The code you have shows that the content of `end_state` is the value of `assigned_roles_after` which this doesn't look like it represents.

Reply:
Yes, I actually copy it from another function. I'm not sure this will represent and I should make a double check with it.